signer/signer.go

```diff
@@ -16,7 +15,7 @@ type SigningService interface {
 
 // CryptoServiceIndex represents a mapping between a service algorithm string
 // and a CryptoService
-type CryptoServiceIndex map[data.KeyAlgorithm]signed.CryptoService
+type CryptoServiceIndex map[string]signed.CryptoService
```

Comment:
@cyli's work on keystores and the follow up work I'm doing to make keys into crypto.Signer implementers will make this type disappear. A single cryptoservice will be able to handle many keystores with many different types of key.